### ISSUE TITLE
Feedback before writing: Use .dockerignore to leave unnecessary artifacts outside of production

### ISSUE THREAD

**Issue by goldbergyoni** [goldbergyoni/nodebestpractices#703]
Context: This is used to share my TOC of a new best practice so I can solicit feedback and ideas before start writing. From my experience, this makes the writing experience much easier and shorter. And fun.

Title: Use .dockerignore to leave unnecessary and sensitive artifacts outside production

**TL&DR:**  The .dockerignore file functionality - filter dev artifacts that are not needed for production; Why is it useful - avoid secrets leak, minimize attack surface, and reduce sizeת increase build time; How practical this is - many developer use the command 'ADD .' which copies everything from the ci/developer folder including .env, .git, .aws and other sensitive files
**Otherwise:** Why is it important - reality tells how frequent secrets are forgotten (many Docker images on the web were found with .git folder and other secrets), this is another safety net
**Simple explanation:** ...
**Advanced tip:** Specifying too much files in the build context (e.g. .git, test-reports) will invalidate the docker build cache and slow down many builds
**Code example:** A common docker ignore, see 1st comment

Thoughts? improvement? Help me to make it better

**Comment by goldbergyoni** [comment]
Example docker ignore:

```
**/node_modules/
**/.git
**/README.md
**/LICENSE
**/.vscode
**/npm-debug.log
**/coverage
**/.env
**/.editorconfig
**/.aws
```

**Comment by kevynb** [comment]
There's also another small advantage: The build is a bit faster, because you are not copying 250mb of data before starting the actual build. I experienced that first hand :)

**Comment by goldbergyoni** [comment]
> There's also another small advantage: The build is a bit faster, because you are not copying 250mb of data before starting the actual build. I experienced that first hand :)

Great point, edited @kevynb

**Comment by kevynb** [comment]
Closing as we merged #721
